### PR TITLE
rsync: update to 3.2.0.

### PR DESCRIPTION
**Pull request by abenson**

### CODE REVIEW

File: srcpkgs/rsync/template [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 short_desc="Remote fast incremental file transfer tool"
 maintainer="Lugubris <lugubris@disroot.org>"
 license="GPL-3.0-only"
 homepage="http://samba.anu.edu.au/rsync/"
 distfiles="http://www.samba.org/ftp/rsync/src/rsync-$version.tar.gz"
-checksum=55cc554efec5fdaad70de921cd5a5eeb6c29a95524c715f3bbf849235b0800c0
+checksum=90127fdfb1a0c5fa655f2577e5495a40907903ac98f346f225f867141424fa25
+
+if [ "$CROSS_BUILD" ] || [ "$XBPS_TARGET_LIBC" = "musl" ]; then
+	configure_args="--disable-simd"
```
Comment (q66) [inline srcpkgs/rsync/template:18]: what's up with this stuff?

**Comment by q66** [comment]
also there's 3.2.1 now

File: srcpkgs/rsync/template (hunk 1 above)
Comment (abenson) [reply srcpkgs/rsync/template:18]: The error was compiler support, but I'm not sure anymore

Comment (abenson) [reply srcpkgs/rsync/template:18]: It looks like they fixed the incompatibilities in 3.2.1, but for cross-build it wants to run a program to determine which SIMD to enable.

Comment (q66) [reply srcpkgs/rsync/template:18]: Just patch it out

Comment (q66) [reply srcpkgs/rsync/template]: Not like that though. I meant to explicitly set the SIMD variant depending on target arch. What you did there is just broken

File: srcpkgs/rsync/patches/fix_cross.patch [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+-
+-	if test x"$CXX_OK" = x"yes"; then
+-	    # AC_MSG_RESULT() is called below.
+-	    SIMD="x86_64"
+-	elif test x"$enable_simd" = x"yes"; then
+-	    { $as_echo "$as_me:${as_lineno-$LINENO}: result: error" >&5
+-$as_echo "error" >&6; }
+-	    as_fn_error $? "The SIMD compilation test failed.
+-Omit --enable-simd to continue without it." "$LINENO" 5
++	if test "$cross_compiling" = yes; then
```
Comment (sgn) [inline srcpkgs/rsync/patches/fix_cross.patch:76]: if not cross_compiling, $build=$target, right?
If I were not wrong, we can simplify to (I don't like unnecessary bashism).
```sh
if test "x${target#x86_64*}" != x; then
     SIMD=x86_64
fi
```

Comment (abenson) [reply srcpkgs/rsync/patches/fix_cross.patch:76]: If not cross-compiling, `$target` is apparently unset.

**Comment by Piraty** [comment]
3.2.2 is out

**Comment by q66** [comment]
well, rebase and we can probably merge it

**Comment by abenson** [comment]
Ah well ac194c11b401027cb1900bde4aa1fdd04e6b551b

**Pull request closed without merging**